File: src/main/java/com/kdopamine/app/domain/word/entity/Chapter.java
package com.kdopamine.app.domain.word.entity;

import com.kdopamine.app.global.entity.BaseEntity;
import com.kdopamine.app.global.entity.SectionType;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "chapters")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder(access = AccessLevel.PRIVATE)
@Getter
public class Chapter extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name_en", nullable = false, unique = true)
    private String nameEn;

    @Column(name = "name_kr", nullable = false, unique = true)
    private String nameKr;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private SectionType type;

    @Column(name = "icon", length = 200)
    private String icon;

    @Column(name = "display_order", nullable = false)
    private Integer displayOrder;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

}

File: k_lingo_front/lib/models/study/chapter.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class Chapter {
  final int id;
  final String nameKr;
  final String nameEn;
  final String description;
  final String icon;

  Chapter({
    required this.id,
    required this.nameKr,
    required this.nameEn,
    required this.description,
    required this.icon,
  });

  // JSON 데이터를 Dart 객체로 변환하는 공장(Factory)
  factory Chapter.fromJson(Map<String, dynamic> json) {
    return Chapter(
      id: json['id'],
      nameKr: json['nameKr'],
      nameEn: json['nameEn'],
      description: json['description'],
      icon: json['icon'],
    );
  }
}